### DISCUSSION TITLE
Localization and multi-lingual support

### DISCUSSION THREAD

**Discussion by jacobwegner** [scaife-viewer/scaife-viewer.org#6] [Ideas]
Within the Scaife Viewer ecosystem, I can see at least two "lenses" of localization to be applied:
1) Localization of the viewer itself (labels, controls, etc). We've done this on other projects using [vue-i18n](https://kazupon.github.io/vue-i18n/).  The existing [scaife-viewer/frontend](https://github.com/scaife-viewer/frontend) would need new hooks, as would [scaife-viewer/scaife-stack](https://github.com/scaife-viewer/scaife-stack)
2) Localization of certain aspects of the data.  An example would be labels for parts of speech annotations, localizing glosses, etc.

For the latter, there are portions of ATLAS where we're tagging metadata with the language value (for labels mostly).  For things like glosses or morphological information, I could see some of the data localization needing to happen within ATLAS (glosses) and some within the viewer (for things like parts of speech...likely done at the "widget" level).

Certain ATLAS queries would then likely need to support a language variable (with a fallback to a default if there isn't a label available).

For both approaches, we should evaluate tools and workflows for managing localization data.  Transifex and BabelEdit are two tools I've personally used in the past and are worth a closer look.

**Comment by jacobwegner** [comment]
@jtauber clarified for me on a conversation that "2." is better expressed as adding "multi-lingual" support to our ATLAS queries.

**Reply by comradekingu** [reply]
@jacobwegner Having used all commonly used tools extensively, I'd say Weblate ranks above in most any category.
There should be more good options to speak of, but there just aren't.

**Comment by jacobwegner** [comment]
Another good use case for multi-lingual support would be the translation of reference widgets, like a dictionary
